Unhandled exception in the UI thread switching contexts

GAPID Version: 0.9.0:0:developer
OS: mac os x x86_64

Switched contexts, got a UI exception:

```
org.eclipse.swt.SWTException: Failed to execute runnable (java.lang.reflect.UndeclaredThrowableException)
	at org.eclipse.swt.SWT.error(Unknown Source)
	at org.eclipse.swt.SWT.error(Unknown Source)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Unknown Source)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Unknown Source)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Unknown Source)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at com.google.gapid.Main$UI.show(Main.java:109)
	at com.google.gapid.Main.main(Main.java:67)
Caused by: java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy7.onStateLoaded(Unknown Source)
	at com.google.gapid.models.ApiState.fireLoadedEvent(ApiState.java:114)
	at com.google.gapid.models.ModelBase.updateSuccess(ModelBase.java:98)
	at com.google.gapid.models.ModelBase$1.onUiThreadSuccess(ModelBase.java:74)
	at com.google.gapid.rpc.UiErrorCallback.onUiThread(UiErrorCallback.java:34)
	at com.google.gapid.rpc.UiErrorCallback.onUiThread(UiErrorCallback.java:25)
	at com.google.gapid.rpc.UiCallback.lambda$onFinish$0(UiCallback.java:46)
	at com.google.gapid.widgets.Widgets.ifNotDisposed(Widgets.java:119)
	at com.google.gapid.widgets.Widgets.lambda$scheduleIfNotDisposed$1(Widgets.java:147)
	at com.google.gapid.widgets.Widgets.lambda$schedule$0(Widgets.java:139)
	at org.eclipse.swt.widgets.RunnableLock.run(Unknown Source)
	... 7 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gapid.util.Events$ListenerCollectionImpl.invoke(Events.java:167)
	... 18 more
Caused by: java.lang.IllegalStateException: Querying children before loaded
	at com.google.common.base.Preconditions.checkState(Preconditions.java:444)
	at com.google.gapid.models.ApiState$Node.getOrCreateChildren(ApiState.java:202)
	at com.google.gapid.models.ApiState$Node.getChildren(ApiState.java:197)
	at com.google.gapid.views.StateView$StateTree$1.getChildNodes(StateView.java:346)
	at com.google.gapid.views.StateView$StateTree$1.getChildNodes(StateView.java:338)
	at com.google.gapid.widgets.LinkifiedTree$ContentProvider.getChildren(LinkifiedTree.java:245)
	at org.eclipse.jface.viewers.AbstractTreeViewer.getRawChildren(AbstractTreeViewer.java:1372)
	at org.eclipse.jface.viewers.TreeViewer.getRawChildren(TreeViewer.java:341)
	at org.eclipse.jface.viewers.StructuredViewer.getFilteredChildren(StructuredViewer.java:907)
	at org.eclipse.jface.viewers.AbstractTreeViewer.getSortedChildren(AbstractTreeViewer.java:621)
	at org.eclipse.jface.viewers.AbstractTreeViewer.lambda$0(AbstractTreeViewer.java:817)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(Unknown Source)
	at org.eclipse.jface.viewers.AbstractTreeViewer.createChildren(AbstractTreeViewer.java:795)
	at org.eclipse.jface.viewers.TreeViewer.createChildren(TreeViewer.java:596)
	at org.eclipse.jface.viewers.AbstractTreeViewer.createChildren(AbstractTreeViewer.java:766)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalExpand(AbstractTreeViewer.java:1636)
	at org.eclipse.jface.viewers.AbstractTreeViewer.setSelectionToWidget(AbstractTreeViewer.java:2519)
	at org.eclipse.jface.viewers.AbstractTreeViewer.setSelectionToWidget(AbstractTreeViewer.java:2989)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1465)
	at org.eclipse.jface.viewers.TreeViewer.preservingSelection(TreeViewer.java:354)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1419)
	at org.eclipse.jface.viewers.AbstractTreeViewer.inputChanged(AbstractTreeViewer.java:1517)
	at com.google.gapid.widgets.VisibilityTrackingTreeViewer.inputChanged(VisibilityTrackingTreeViewer.java:76)
	at org.eclipse.jface.viewers.ContentViewer.setInput(ContentViewer.java:286)
	at org.eclipse.jface.viewers.StructuredViewer.setInput(StructuredViewer.java:1696)
	at com.google.gapid.widgets.LinkifiedTree.setInput(LinkifiedTree.java:148)
	at com.google.gapid.views.StateView.onStateLoaded(StateView.java:216)
	... 23 more
```

--- a/gapic/src/main/com/google/gapid/widgets/LinkifiedTree.java
+++ b/gapic/src/main/com/google/gapid/widgets/LinkifiedTree.java
@@ -145,6 +145,10 @@ public abstract class LinkifiedTree<T, F> extends Composite {
   }
 
   public void setInput(T root) {
+    // Clear the selection, since we handle maintaining the selection ourselves and so
+    // don't want JFace's selection preserving, as it appears to be broken on input
+    // change (see https://github.com/google/gapid/issues/1264)
+    setSelection(null);
     viewer.setInput(root);
     if (root != null && viewer.getTree().getItemCount() > 0) {
       viewer.getTree().setSelection(viewer.getTree().getItem(0));

